components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/validationhandler/impl/RoleBasedScopeValidationHandler.java

```diff
@@ -99,9 +101,12 @@ public List<String> validateScopes(List<String> requestedScopes, List<String> ap
             List<String> filteredScopes = appAuthorizedScopes.stream().filter(associatedScopes::contains)
                     .collect(Collectors.toList());
             return requestedScopes.stream().filter(filteredScopes::contains).collect(Collectors.toList());
+        } catch (IdentityOAuth2ClientException e) {
+                throw new ScopeValidationHandlerClientException("Error while validating scope with RBAC Scope " +
+                        "Validation handler", e);
```

Comment:
Since we have followed the path on catching the parent exception and differentiating the client exception through `instance of` in [1], shall we follow a consistence  approach in both places?

[1] - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2475/files#diff-9e3415cbfce720381cde67a3ee35137800f2c6634cf503faee1bb4fe45b0635eR212-R215

Reply:
This is now updated.